Workload-Security/Support/ws-ssm-support-automation.yaml

```diff
@@ -0,0 +1,191 @@
+  The output of this tool can be provided to Trend Micro Support on case creation.
+
+  ---
+
+  ## Requirements for use.
+
+  ### SSM Automation Operations IAM Role
+  - Create an IAM Role for SSM Automations to Assume to execute document. See example [SSM-Policy](https://github.com/JustinDPerkins/TrendCloudOne-SupportCollection/blob/main/Workload-Security/aws/ssm-iam-example-policy.json)
+  - The IAM Role requires SSM to be a Trusted Entity.
+
+  ### Instance Requirements:
+  - Requires Workload Security Agent to be deployed on instance.
+  - Requires SSM Agent to be Installed and Running. See [here](https://docs.aws.amazon.com/systems-manager/latest/userguide/ssm-agent.html)
+  - The EC2 Instance requires an IAM Role with SSMManagedCore permissions.
+  - The EC2 will need an IAM Role with S3:PutObject permissions to upload diagnostic package to S3.
+  - The Instance will require the AWS CLI to be installed. See [here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
+
+  ---
+
+  ### Limitations:
+  - Same Account/Region
+
+  ---
+
+  ### What is supported?
+  - Linux OS
+  - Windows OS
+
+  ---
+  ## How Can this tool be used?
+
+  ### Via AWS Console?
+  Systems Manager > Documents > All Documents > "Trend-WorkloadSecurity-SupportCollectionTool"
+  - Provide the ARN value of the SSM Automation Operation IAM Role will assume.(SSM Trusted Entity)
+  - Provide a single or comma seperated list of Instance ID. (i-1234567890,...,...)
+  - Provide the Name of the S3 bucket to upload the diagnostic package to.(EC2 will need permissions to this bucket to put objects)
+
+  ### Via AWS CLI:
+  - Linux CLI:
+
+  ```
+  aws ssm start-automation-execution --document-name "Trend-WorkloadSecurity-SupportCollectionTool" --document-version "\$DEFAULT" --parameters '{"AutomationAssumeRole":["<SSM IAM Automation Role ARN Here>"],"InstanceIds":["<Instance-ID-Here>"],"S3BucketName":["<Bucket-4-Artifact-Name-Here>"]}' --region us-east-1
+  111
+  ```
+
+
+schemaVersion: '0.3'
+assumeRole: '{{ AutomationAssumeRole }}'
+parameters:
+  AutomationAssumeRole:
+    type: String
+    description: (Required) The ARN of the role that allows Automation to perform the actions on your behalf
+  InstanceIds:
+    type: StringList
+    description: EC2 instance ID. Comma Seperated ID's if multiple
+  S3BucketName:
+    type: String
+    description: S3 bucket name to upload the diagnostic package
+mainSteps:
+  - name: GetSSMInstanceInformation
+    action: 'aws:executeAwsApi'
+    inputs:
+      Service: ssm
+      Api: DescribeInstanceInformation
+      Filters:
+        - Key: InstanceIds
+          Values:
+            - '{{ InstanceIds }}'
+    outputs:
+      - Name: instances
+        Selector: $.InstanceInformationList
+        Type: MapList
+  - name: CompileInstanceList
+    action: 'aws:executeScript'
+    inputs:
+      Runtime: python3.7
```

Comment:
python3.7 is EOL, the Security support end in 3 weeks.

Reply:
Updated to 3.8